src/combinatorics/quiver/iso.lean

```diff
@@ -0,0 +1,187 @@
+/-
+Copyright (c) 2023 Adam Topaz, Rémi Bottinelli. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz, Rémi Bottinelli
+-/
+import combinatorics.quiver.basic
+import combinatorics.quiver.cast
+import logic.equiv.basic
+import tactic.nth_rewrite
+/-!
+# Iso
+
+Isomorphisms of quivers, defined as pairs of prefunctors that compose to the identities.
+
+## Main definitions
+
+* For quivers `U` and `V`, `iso U V` is the type of isomorphisms between `U` and `V`, with
+  associated `iso.refl`, `iso.symm`, and `iso.trans` definitions.
+* `iso.of_bijective` is the isomorphism defined by a prefunctor that is bijective on vertices and
+  arrows.
+
+## Notation
+
+* `U ≃q V` is notation for `iso U V`
+
+-/
+
+universes u v w z
+
+namespace quiver
+
+/--
+An isomorphism of quivers is given by a pair of prefunctors whose two composites are the identities.
+-/
+structure iso (U V : Type*) [quiver.{u+1} U] [quiver.{v+1} V] extends prefunctor U V :=
+(inv_prefunctor : V ⥤q U)
+(left_inv : to_prefunctor ⋙q inv_prefunctor = 𝟭q _)
+(right_inv : inv_prefunctor ⋙q to_prefunctor = 𝟭q _)
+
+infix ` ≃q `:60 := iso
+
+variables {U V W Z : Type*} [quiver.{u+1} U] [quiver.{v+1} V] [quiver.{w+1} W] [quiver.{z+1} Z]
+
+instance : has_coe (iso U V) (prefunctor U V) := ⟨iso.to_prefunctor⟩
+
+/--
+Two isomorphisms are equal iff their `to_prefunctor` and `inv_prefunctor` agree.
+Not tagged `@[ext]` because `to_prefunctor_ext` will be.
+-/
+lemma iso.ext (φ ψ : iso U V)
+  (hto : φ.to_prefunctor = ψ.to_prefunctor) (hinv : φ.inv_prefunctor = ψ.inv_prefunctor) : φ = ψ :=
```

Comment:
This should be stated in terms of `⇑φ` and `⇑φ⁻¹`, and you'll have to set up the `fun_like`/`equiv_like` stuff.

Reply:
Actually, I'm not sure I understand. `fun_like` is described as a typeclass when your type maps injectively to `A -> B`, and `equiv_like` when it maps injectively to `A ~= B`. Neither is the case here, so what kind of instance should I have?

Reply:
Hmm, maybe nothing indeed.